src/middlewares/auth.js

```diff
@@ -38,14 +41,51 @@ const auth = {
           error: 'Failed to authenticate token.',
         });
       }
+
+      const user = await User.findOne({ where: { id: decoded.payload.id } });
+      // if (!user) {
+      //   return res.status(401).json({
+      //     status: 'error',
+      //     error: 'This user does not exist.',
+      //   });
+      // }
+      req.currentUser = user;
       return next();
     } catch (error) {
       return res.status(500).json({
         status: 'error',
-        error: 'Internal Server Error',
+        error: ('Internal Server Error'),
       });
     }
-  }
+  },
+
+  async verifyManager(req, res, next) {
```

Comment:
Please how is this code different from other methods that verify token?

Reply:
I needed to distinct it because I will be passing it as a middleware only for super_admin/manager protected routes

Reply:
You can use rhe same middleware. If you signed the role, you can still get it using the same function.

Reply:
Okay. I will implement that